js/src/builtin/streams/MiscellaneousOperations.cpp

```diff
@@ -30,6 +30,54 @@ using JS::MutableHandle;
 using JS::ToNumber;
 using JS::Value;
 
+// https://streams.spec.whatwg.org/#transfer-array-buffer
+// As some parts of the specifcation want to use the abrupt completion value,
+// this function may leave a pending exception if it returns nullptr.
```

Comment:
According to the spec (https://streams.spec.whatwg.org/#transfer-array-buffer), the call to `DetachArrayBuffer`:

> This will throw an exception if O has an [[ArrayBufferDetachKey]] that is not undefined, such as a [WebAssembly.Memory](https://webassembly.github.io/spec/js-api/#memory)'s [buffer](https://webassembly.github.io/spec/js-api/#dom-memory-buffer). [[WASM-JS-API-1]](https://streams.spec.whatwg.org/#biblio-wasm-js-api-1)

That's the only path that returns null in this function, so I believe we are good so long as `DetachArrayBuffer` behaves.